sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/ByteBufWriteSubscriber.java

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.netty.implementation;
+
+import io.netty.buffer.ByteBuf;
+import org.reactivestreams.Subscriber;
+import org.reactivestreams.Subscription;
+import reactor.core.publisher.MonoSink;
+import reactor.core.publisher.Operators;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+
+/**
+ * Subscriber that writes a stream of {@link ByteBuf ByteBufs} to a {@link WritableByteChannel}.
+ */
+@SuppressWarnings("ReactiveStreamsSubscriberImplementation")
+public class ByteBufWriteSubscriber implements Subscriber<ByteBuf> {
+    private final WritableByteChannel channel;
+    private final MonoSink<Void> emitter;
+
+    private Subscription subscription;
+
+    public ByteBufWriteSubscriber(WritableByteChannel channel, MonoSink<Void> emitter) {
```

Comment:
Can we take a consumer to do the `write` operation instead of `channel` and consolidate these two subscribers to reduce duplication?

Reply:
Yeah I can do that